src/luasql.c

```diff
@@ -131,3 +131,33 @@ LUASQL_API void luasql_set_info (lua_State *L) {
 	lua_pushliteral (L, "LuaSQL 2.3.0");
 	lua_settable (L, -3);
 }
+
+/*
+** Pulls an optional string value from the table at idx
+*/
+LUASQL_API const char* luasql_table_optstring(lua_State *L, int idx, const char* name, const char* def) {
+	const char* res = NULL;
+
+	lua_pushstring(L, name);
+	lua_gettable(L, idx);
+
+	res = lua_tostring(L, -1);
+	lua_pop(L, 1);
+
+	return (res != NULL) ? res : def;
+}
+
+/*
+** Pulls an optional number value from the table at idx
+*/
+LUASQL_API lua_Number luasql_table_optnumber(lua_State *L, int idx, const char* name, lua_Number def) {
+	lua_Number res = 0;
+
+	lua_pushstring(L, name);
+	lua_gettable(L, idx);

```

Comment:
Hi Peter

> @tomasguisasola [2] so the compatibility with 5.0 can be removed?
Yes.

Regards,
Tomás

On 2016-07-05 10:26, Peter Melnichenko wrote:
> In src/luasql.c [1]:
> 
>> + lua_gettable(L, idx);
>> +
>> + res = lua_tostring(L, -1);
>> + lua_pop(L, 1);
>> +
>> + return (res != NULL) ? res : def;
>> +}
>> +
>> +/*
>> +** Pulls an optional number value from the table at idx
>> +*/
>> +LUASQL_API lua_Number luasql_table_optnumber(lua_State *L, int idx,
> const char* name, lua_Number def) {
>> + lua_Number res = 0;
>> +
>> + lua_pushstring(L, name);
>> + lua_gettable(L, idx);
> 
> @tomasguisasola [2] so the compatibility with 5.0 can be removed?
> Challenges are fun, but so is replacing boilerplate
> lua_pushliteral/lua_pushvalue/lua_settable calls with lua_setfield,
> etc
> 
> --
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub [3], or mute the
> thread [4].
> 
> 	*
> 
> Links:
> ------
> [1] 
> https://github.com/keplerproject/luasql/pull/54#discussion_r69561897
> [2] https://github.com/tomasguisasola
> [3]
> https://github.com/keplerproject/luasql/pull/54/files/b4afdfb93567f8ffaa268ed63115bb9672a106aa#r69561897
> [4]
> https://github.com/notifications/unsubscribe/AAIA7RQ7RhQ7s497Z87XUnyGmAB_Bk7rks5qSluNgaJpZM4JEQdV